gulpfile.js/changelog.js

```diff
@@ -0,0 +1,380 @@
+"use strict";
+
+const { src, dest } = require('gulp');
+
+const replace = require('gulp-replace');
+const pump = require('pump');
+const git = require('simple-git/promise')(__dirname);
+
+const { changelog } = require('./paths');
+
+
+const ISSUE_RE = /#(\d+)(?![\d\]])/g;
+const ISSUE_SUB = '[#$1](https://github.com/PrismJS/prism/issues/$1)';
+
+function linkify(cb) {
+	return pump([
+		src(changelog),
+		replace(ISSUE_RE, ISSUE_SUB),
+		replace(
+			/\[[\da-f]+(?:, *[\da-f]+)*\]/g,
+			m => m.replace(/([\da-f]{7})[\da-f]*/g, '[`$1`](https://github.com/PrismJS/prism/commit/$1)')
+		),
+		dest('.')
+	], cb);
+}
+
+/**
+ * Creates an array which iterates its items in the order given by `compareFn`.
+ *
+ * The array may not be sorted at all times.
+ *
+ * @param {(a: T, b: T) => number} compareFn
+ * @returns {T[]}
+ * @template T
+ */
+function createSortedArray(compareFn) {
+	const a = [];
+
+	a['sort'] = function () {
+		return Array.prototype.sort.call(this, compareFn);
+	};
+	a[Symbol.iterator] = function* () {
+		const copy = [];
+		for (let i = 0; i < this.length; i++) {
+			copy.push(this[i]);
+		}
```

Comment:
`this.slice()` is less code.

Reply:
I wasn't sure whether `slice` uses the iterator under the hood but apparently, it doesn't.
So, will do.